pkg/pipeline/transform/transform_network.go

```diff
@@ -40,9 +41,23 @@ type Network struct {
 }
 
 func (n *Network) Transform(input []config.GenericMap) []config.GenericMap {
+	entriesChan := make(chan config.GenericMap, len(input))
+	var wg sync.WaitGroup
+	wg.Add(len(input))
+
+	for _, e := range input {
+		go func(entry config.GenericMap) {
+			defer wg.Done()
+			outputEntry := n.TransformEntry(entry)
+			entriesChan <- outputEntry
+		}(e)
```

Comment:
I'm afraid that the goroutine creation/management and channel management costs could repel any parallelization improvement. TransformEntry is a short process, and spawning 10K goroutines to hold each of them in parallel could not be effective.

I'd rather divide the work in less tasks carrying a bigger portion of the work. E.g. carry a subslice on each goroutine:

```go
outputEntries := make([]config.GenericMap, len(input))
parallelTasks := runtime.GOMAXPROCS(0) // or halve it, or do user-configurable 
for t := 0; t < parallelTasks ; t++ {
    si := t * len(input) / parallelTasks
    ei := (t + 1) * len(input) / parallelTasks
    go func() {
        defer wg.Done()
        for i, entry := range input[si: ei] {
            outputEntries[i + si] = n.TransformEntry(entry)
        }
    }()
}
```

Reply:
("10K goroutines", thinking in the use case of one of our users)

Reply:
yeah ... "goroutines are cheap", but maybe not THAT cheap :)

Reply:
anyway, the need for parallel processing was assuming there was blocking calls on the kube API, but if that's not the case (still unclear to me), we can forget this